packages/account/src/account.test.ts

```diff
@@ -23,7 +23,62 @@ describe('Account', () => {
   });
 
   async function setupTestProvider(providerOptions = {}) {
-    const { provider, cleanup } = await setupTestProviderAndWallets({ providerOptions });
+    const { provider, cleanup } = await setupTestProviderAndWallets({
+      providerOptions,
+      nodeOptions: {
+        snapshotConfig: {
+          stateConfig: {
+            coins: [
+              {
+                tx_id: '0x0000000000000000000000000000000000000000000000000000000000000002',
+                output_index: 0,
+                tx_pointer_block_height: 0,
+                tx_pointer_tx_idx: 0,
+                owner: '0x09c0b2d1a486c439a87bcba6b46a7a1a23f3897cc83a94521a96da5c23bc58db',
+                amount: bn('0xFFFFFFFFFFFFFFFF', 'hex'),
+                asset_id: '0xf8f8b6283d7fa5b672b530cbb84fcccb4ff8dc40f8176ef4544ddb1f1952ad07',
+              },
```

Comment:
Would it be worthwhile having a `TestCoin` helper function, similar to the `TestMessage`.

We could have a static `many` function which could generate multiple coins.

```ts
const coins: Coin[] = TestCoin.many({
  owner: '0x09c0b2d1a486c439a87bcba6b46a7a1a23f3897cc83a94521a96da5c23bc58db'
}, 4);
```

Reply:
Great point but to be fair, this PR already closes two issues and has a lot of changes. Do you mind if I file a follow-up issue for this and get this done in another PR?

Reply:
https://github.com/FuelLabs/fuels-ts/issues/3445